Fix golint failures

Anyone itching to understand the code and try fixing things in Kubernetes, please help fixing the lint failures

Step 1: Remove one or two packages (related packages) from the list in - https://github.com/kubernetes/kubernetes/blame/master/hack/.golint_failures (note: do not change generated files, the name of defaulting functions, or the name of conversion functions... packages failing linting due to issues in those files will need to be left alone)
Step 2: Run `hack/verify-golint.sh` and look at the logged errors
Step 3: Fix the code to ensure that `hack/verify-golint.sh` runs cleanly.
Step 4: Run `hack/update-bazel.sh` and `hack/update-gofmt.sh` to make sure we fix any problems with gofmt or bazel.
Step 5: File a PR with the change
Step 6: Go to step #1 :)

Please use the tracking spreadsheet [here](https://docs.google.com/spreadsheets/d/1VhU6zCk-vaAnbu_XkgsIq5I7jVnezUquWBMhcIN6AJM/edit#gid=0) For everyone who is going to contribute, 
- please append the packages you picked at the end.
- For existing contributor, please make sure this list is update to date

I keep getting this golint verify error on `master` and was wondering if others saw it:
```
hack/verify-golint.sh
installing golint from vendor
Errors from golint:
test/integration/apiserver/admissionwebhook/broken_webhook_test.go:153:20: should omit type admissionregistrationv1beta1.FailurePolicyType from declaration of var failurePolicy; it will be inferred from the right-hand side

Please review the above warnings. You can test via "golint" and commit the result.
If the above warnings do not make sense, you can exempt this package from golint
checking by adding it to hack/.golint_failures (if your reviewer is okay with it).
```
But when I added that package to the golint failures in my PR, my reviewer said it isn't necessary? What does that mean? What did I miss? Should I use a different kind of verify?
cf: https://github.com/kubernetes/kubernetes/pull/77033#discussion_r278377666

--- a/hack/.golint_failures
+++ b/hack/.golint_failures
@@ -593,12 +593,8 @@ staging/src/k8s.io/sample-apiserver/pkg/admission/plugin/banflunder
 staging/src/k8s.io/sample-apiserver/pkg/admission/wardleinitializer
 staging/src/k8s.io/sample-apiserver/pkg/apis/wardle
 staging/src/k8s.io/sample-apiserver/pkg/apis/wardle/v1alpha1
-staging/src/k8s.io/sample-apiserver/pkg/apiserver
-staging/src/k8s.io/sample-apiserver/pkg/cmd/server
 staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/fischer
 staging/src/k8s.io/sample-apiserver/pkg/registry/wardle/flunder
-staging/src/k8s.io/sample-controller/pkg/apis/samplecontroller
-staging/src/k8s.io/sample-controller/pkg/apis/samplecontroller/v1alpha1
 test/e2e
 test/e2e/auth
 test/e2e/autoscaling

--- a/staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go
+++ b/staging/src/k8s.io/sample-apiserver/pkg/apiserver/apiserver.go
@@ -33,7 +33,10 @@ import (
 )
 
 var (
+	// Scheme defines methods for serializing and deserializing API objects.
 	Scheme = runtime.NewScheme()
+	// Codecs provides methods for retrieving codecs and serializers for specific
+	// versions and content types.
 	Codecs = serializer.NewCodecFactory(Scheme)
 )
 
@@ -55,10 +58,12 @@ func init() {
 	)
 }
 
+// ExtraConfig holds custom apiserver config
 type ExtraConfig struct {
 	// Place you custom config here.
 }
 
+// Config defines the config for the apiserver
 type Config struct {
 	GenericConfig *genericapiserver.RecommendedConfig
 	ExtraConfig   ExtraConfig
@@ -74,8 +79,8 @@ type completedConfig struct {
 	ExtraConfig   *ExtraConfig
 }
 
+// CompletedConfig embeds a private pointer that cannot be instantiated outside of this package.
 type CompletedConfig struct {
-	// Embed a private pointer that cannot be instantiated outside of this package.
 	*completedConfig
 }
 

--- a/staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go
+++ b/staging/src/k8s.io/sample-apiserver/pkg/cmd/server/start.go
@@ -41,6 +41,7 @@ import (
 
 const defaultEtcdPathPrefix = "/registry/wardle.kubernetes.io"
 
+// WardleServerOptions contains state for master/api server
 type WardleServerOptions struct {
 	RecommendedOptions *genericoptions.RecommendedOptions
 
@@ -49,6 +50,7 @@ type WardleServerOptions struct {
 	StdErr                io.Writer
 }
 
+// NewWardleServerOptions returns a new WardleServerOptions
 func NewWardleServerOptions(out, errOut io.Writer) *WardleServerOptions {
 	o := &WardleServerOptions{
 		RecommendedOptions: genericoptions.NewRecommendedOptions(
@@ -92,12 +94,14 @@ func NewCommandStartWardleServer(defaults *WardleServerOptions, stopCh <-chan st
 	return cmd
 }
 
+// Validate validates WardleServerOptions
 func (o WardleServerOptions) Validate(args []string) error {
 	errors := []error{}
 	errors = append(errors, o.RecommendedOptions.Validate()...)
 	return utilerrors.NewAggregate(errors)
 }
 
+// Complete fills in fields required to have valid data
 func (o *WardleServerOptions) Complete() error {
 	// register admission plugins
 	banflunder.Register(o.RecommendedOptions.Admission.Plugins)
@@ -108,6 +112,7 @@ func (o *WardleServerOptions) Complete() error {
 	return nil
 }
 
+// Config returns config for the api server given WardleServerOptions
 func (o *WardleServerOptions) Config() (*apiserver.Config, error) {
 	// TODO have a "real" external address
 	if err := o.RecommendedOptions.SecureServing.MaybeDefaultWithSelfSignedCerts("localhost", nil, []net.IP{net.ParseIP("127.0.0.1")}); err != nil {
@@ -138,6 +143,7 @@ func (o *WardleServerOptions) Config() (*apiserver.Config, error) {
 	return config, nil
 }
 
+// RunWardleServer starts a new WardleServer given WardleServerOptions
 func (o WardleServerOptions) RunWardleServer(stopCh <-chan struct{}) error {
 	config, err := o.Config()
 	if err != nil {

--- a/staging/src/k8s.io/sample-controller/pkg/apis/samplecontroller/register.go
+++ b/staging/src/k8s.io/sample-controller/pkg/apis/samplecontroller/register.go
@@ -16,6 +16,7 @@ limitations under the License.
 
 package samplecontroller
 
+// GroupName is the group name used in this package
 const (
 	GroupName = "samplecontroller.k8s.io"
 )

--- a/staging/src/k8s.io/sample-controller/pkg/apis/samplecontroller/v1alpha1/register.go
+++ b/staging/src/k8s.io/sample-controller/pkg/apis/samplecontroller/v1alpha1/register.go
@@ -38,8 +38,10 @@ func Resource(resource string) schema.GroupResource {
 }
 
 var (
+	// SchemeBuilder initializes a scheme builder
 	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-	AddToScheme   = SchemeBuilder.AddToScheme
+	// AddToScheme is a global function that registers this API group & version to a scheme
+	AddToScheme = SchemeBuilder.AddToScheme
 )
 
 // Adds the list of known types to Scheme.

